Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Favorites/AssetBrowserFavoritesView.cpp

```diff
@@ -231,6 +234,49 @@ namespace AzToolsFramework
             }
         }
 
+        
+        void AssetBrowserFavoritesView::SelectionChanged(const QItemSelection& selected, [[maybe_unused]] const QItemSelection& deselected)
+        {
+            m_selectionChangedSinceLastClick = true;
+            NotifySelection(selected);
+        }
+
+        void AssetBrowserFavoritesView::NotifySelection(const QItemSelection& selected)
+        {
+            // if we select 1 thing, give the previewer a chance to view it.  Favorites is a cell-by-cell selection
+            // so it won't select an entire row, ie, every selection will be its own row so its not necessary to count rows, as that will
+            // be the same as the count of the selected indexes.
+            QModelIndexList selectedIndexes = selected.indexes();
+            if (selectedIndexes.size() == 1)
```

Comment:
I would go for early returns instead of nesting

Reply:
The reason I did  not here, is that we actually want it to invoke `AssetBrowserPreviewRequestBus::Broadcast(&AssetBrowserPreviewRequest::ClearPreview);` when it didn't hit the positive condition.

I always prefer an Early Return type style, but only when its a trivial return, that is
```cpp
if (some condition)
{  
   return;
}
```

I like it a lot less when the return case has to have extra lines (besides return) in there.

Reply:
as per above, I'll leave an extra comment in there, but I'll keep the existing structure.

Reply:
Ok no problem :)